Automattic/jetpack
# Social Previews | Ensure the correct image/media is used.

Fixes 1203820933396971-as-1204615220483685/f

Related: https://github.com/Automattic/wp-calypso/pull/77055

## Proposed changes:

* Update SIG config to save the token to post meta when generated.
* Give preference to SIG image when available for previews
* Update Facebook, Twitter and LinkedIn previews to handle custom media

### Other information:


## Jetpack product discussion


pdrWKz-Ka-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* In Jetpack, run `jetpack build plugins/jetpack`
* Start writing a new post with a JT site with Social Advanced plan/features enabled
* Add a Featured Image to the post
* Click Jetpack icon on the top of the sidebar
* Under "Social Previews", click on "Preview"
* Confirm that the Facebook, Twitter and LinkedIn previews show similar "Your post" and "Link preview" sections
* Confirm that the featured image is shown in both the previews
* Close the modal
* Under "Social Image Generator", turn ON "Enable Social Image" option 
* Open the Preview modal again
* Confirm that SIG image is now shown instead of the Featured image
* Close the modal and turn ON "Share as a social post" option
* Open the preview modal again
* Confirm that "Your post" section now shows the media (image) instead of the preview to a post
* Confirm that the "Link preview" shows what it showed before
* Turn OFF "Enable Social Image" option 
* Under "Media" choose a media file (image/video)
* Open the preview modal again
* Confirm that "Your post" section now shows the custom media instead of featured image and SIG image
* Turn ON "Enable Social Image" option again
* Open the preview modal
* Confirm that SIG image is preferred over custom media in the preview

BEFORE

https://github.com/Automattic/wp-calypso/assets/18226415/b758a1cf-5d0e-4772-a2e1-88bb8a59dae2

AFTER

https://github.com/Automattic/wp-calypso/assets/18226415/4dbcc482-cb35-4eef-a5cf-82132dc520f5

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1204615220483685
